bemuse/src/music-collection/preprocessCollection.js

```diff
@@ -1,10 +1,11 @@
-import u from 'updeep'
+import produce from "immer"
 
 export const preprocessCollection = u({
   songs: u.map(preprocessSong),
 })
 
 function preprocessSong(song) {
+  const useImmerMap = new Map()
```

Comment:
@dtinth I am having trouble using immer maps in this part.

Reply:
You don’t need to use a map — instead, rewrite the logic to match the original behavior. Here’s an example (untested).

```js
  return produce(song, draft => {
    if (song.chart_names) {
      for (const chart of song.charts) {
        const name = song.chart_names[chart.file]
        if (!name) continue
        chart.info.subtitles.push(name)
      }
    }
  })
```

Reply:
Ok got it.